### DISCUSSION TITLE
Problems trying to use finite state machine definiton

### DISCUSSION THREAD

**Discussion by kmerbcn** [OpenModelica/OpenModelica#12611] [General]
Hello

I tried to solve the following model based on a finite state machine. Nevertheless it seems the code has 2 variables and 1 equation. How can I write the model of a finite state machine? In theory, I followed the help and the standard.
```modelica
model PruebaEstados
  inner Integer i(start=0);
  
  block State1
    outer output Integer i;
   equation
    i = previous(i) + 2;
  end State1;
  
  block State2
    outer output Integer i;
   equation
    i = previous(i) - 1;
  end State2;
  
  State1 state1;
  State2 state2;
  
equation
  initialState(state1);
  transition(state1, state2, i > 10, immediate =false ,reset=true , synchronize =false ,priority =1);
  transition(state2, state1, i < 1, immediate =false ,reset=true , synchronize =false ,priority =1);

end PruebaEstados;
```

**Comment by casella** [comment]
@perost, what is the status with state machine support in the new frontend?

**Comment by kmerbcn** [comment]
Hello Francesco

Thanks for repplying, It seems any of us is going to bed now.
I followed the Omedit documentation as stated here :
https://openmodelica.org/doc/OpenModelicaUsersGuide/1.17/omedit.html#state-machines

I created two states as the documentation says. Finally a model containing these models and the inner Integer variable i. Inside of equations I wrote the transitions stamenets and finally in Options -> Simulation I wrote in the additional translation flag : --std=latest , as stated in the documentation. I hope I can do this simple model.

Thanks to all

**Reply by kmerbcn** [reply]
![image](https://github.com/OpenModelica/OpenModelica/assets/62482095/e4f15ab2-6a91-4ef9-81e7-9595bfd92928)

**Comment by casella** [comment]
Try setting additional translation flag `--nonewInst`

**Reply by kmerbcn** [reply]
Hello Francesco


I tried without a solution

![image](https://github.com/OpenModelica/OpenModelica/assets/62482095/4f54bf6b-5bb8-4084-9a67-20f01a82f479)


It seems I get translation error. This is very strange. I wonder why such a simple model does not work on my laptop

**Comment by casella** [comment]
So far we haven't focus much on state machines. Are you just playing around with them or are you seriously interested? It may be the time to delve into it more thoroughly.

**Comment by kmerbcn** [comment]
For the moment, I overcame the problem by looking at other examples with when clauses.
Of course, I would be happy to have, since I am doing a control of a power inverter and i need to control the drivers of 2 bridges of MOSFETs. In this way, I can be very close to the way I would program the control in a microcontroller or PFGA. I hope you understand my point. In any case, Modelica is a nice tool. In any case, I would like to know what I am doing wrong with it.

**Reply by casella** [reply]
OK, maybe someone else with more experience on state machines could answer 😃
